src/workerd/server/workerd-api.c++

```diff
@@ -157,17 +159,24 @@ struct WorkerdApi::Impl final {
 
   Impl(jsg::V8System& v8System,
       CompatibilityFlags::Reader featuresParam,
-      IsolateLimitEnforcer& limitEnforcer,
-      kj::Own<jsg::IsolateObserver> observer,
+      kj::Own<IsolateLimitEnforcer> limitEnforcerParam,
+      kj::Own<IsolateObserver> observerParam,
       api::MemoryCacheProvider& memoryCacheProvider,
       const PythonConfig& pythonConfig = defaultConfig,
       kj::Maybe<kj::Own<jsg::modules::ModuleRegistry>> newModuleRegistry = kj::none)
       : features(capnp::clone(featuresParam)),
         maybeOwnedModuleRegistry(kj::mv(newModuleRegistry)),
-        jsgIsolate(
-            v8System, Configuration(*this), kj::mv(observer), limitEnforcer.getCreateParams()),
+        limitEnforcer(kj::mv(limitEnforcerParam)),
+        observer(kj::atomicAddRef(*observerParam)),
+        jsgIsolate(v8System,
+            Configuration(*this),
+            kj::mv(observerParam),
+            limitEnforcer->getCreateParams()),
         memoryCacheProvider(memoryCacheProvider),
-        pythonConfig(pythonConfig) {}
+        pythonConfig(pythonConfig) {
+    jsgIsolate.runInLockScope(
+        [&](JsgWorkerdIsolate::Lock& lock) { limitEnforcer->customizeIsolate(lock.v8Isolate); });
```

Comment:
In the previous version of the code this is run under `jsg::runInV8Stack`, I assume doing this here under `runInLockScope` is equivalent.

Reply:
Yeah, it's a utility function that translates to the same thing :)